src/services/templateService.ts

```diff
@@ -69,18 +73,63 @@ export const generateVersion = async (
     comment,
     active: true,
   });
+
+  const created = await versionedTemplate.create(context);
+
+  // If the version was successfully created and there are no errors
+  if (created && (!created.errors || (Array.isArray(created.errors) && created.errors.length === 0))) {
+    const sections = await Section.findByTemplateId('generateTemplateVersion', context, template.id);
+
+    try {
+      let allSectionsWereVersioned = true;
+
+      sections.forEach(async (section) => {
+        const sectionInstance = new Section({
```

Comment:
Need to create new instances of section here because "generateSectionVersion" calls on ".create()" method from Section class.